### ISSUE TITLE
wis2box sometimes fails to start properly

### ISSUE THREAD

**Issue by C-Loftus** [cgs-earth/wis2box#9]
Hard to tell when this is occuring and why. Sometimes it will be unable to generate the OpenAPI document or take too long and just go into a restart loop in docker

```
wis2box-api             | ERROR: ERROR: OpenAPI document could not be generated
loki                    | level=info ts=2024-12-04T14:55:59.055319968Z caller=table_manager.go:171 msg="uploading tables"
loki                    | level=info ts=2024-12-04T14:55:59.059663134Z caller=table.go:335 msg="uploading table index_20061"
loki                    | level=info ts=2024-12-04T14:55:59.059708843Z caller=table.go:362 msg="finished uploading table index_20061"
loki                    | level=info ts=2024-12-04T14:55:59.059728009Z caller=table.go:420 msg="cleaning up unwanted dbs from table index_20061"
wis2box                 | Traceback (most recent call last):
wis2box                 |   File "/usr/local/lib/python3.9/site-packages/urllib3/connection.py", line 199, in _new_conn
wis2box                 |     sock = connection.create_connection(
wis2box                 |   File "/usr/local/lib/python3.9/site-packages/urllib3/util/connection.py", line 60, in create_connection
wis2box                 |     for res in socket.getaddrinfo(host, port, family, socket.SOCK_STREAM):
wis2box                 |   File "/usr/local/lib/python3.9/socket.py", line 954, in getaddrinfo
wis2box                 |     for res in _socket.getaddrinfo(host, port, family, type, proto, flags):
wis2box                 | socket.gaierror: [Errno -2] Name or service not known
wis2box                 | 
wis2box                 | The above exception was the direct cause of the following exception:
wis2box                 | 
wis2box                 | Traceback (most recent call last):
wis2box                 |   File "/usr/local/lib/python3.9/site-packages/urllib3/connectionpool.py", line 789, in urlopen
wis2box                 |     response = self._make_request(
wis2box                 |   File "/usr/local/lib/python3.9/site-packages/urllib3/connectionpool.py", line 495, in _make_request
wis2box                 |     conn.request(
wis2box                 |   File "/usr/local/lib/python3.9/site-packages/urllib3/connection.py", line 441, in request
wis2box                 |     self.endheaders()
wis2box                 |   File "/usr/local/lib/python3.9/http/client.py", line 1280, in endheaders
wis2box                 |     self._send_output(message_body, encode_chunked=encode_chunked)
wis2box                 |   File "/usr/local/lib/python3.9/http/client.py", line 1040, in _send_output
wis2box                 |     self.send(msg)
wis2box                 |   File "/usr/local/lib/python3.9/http/client.py", line 980, in send
wis2box                 |     self.connect()
wis2box                 |   File "/usr/local/lib/python3.9/site-packages/urllib3/connection.py", line 279, in connect
wis2box                 |     self.sock = self._new_conn()
wis2box                 |   File "/usr/local/lib/python3.9/site-packages/urllib3/connection.py", line 206, in _new_conn
wis2box                 |     raise NameResolutionError(self.host, self, e) from e
wis2box                 | urllib3.exceptions.NameResolutionError: <urllib3.connection.HTTPConnection object at 0xffff9c22eca0>: Failed to resolve 'wis2box-api' ([Errno -2] Name or service not known)
wis2box                 | 
wis2box                 | The above exception was the direct cause of the following exception:
wis2box                 | 
wis2box                 | Traceback (most recent call last):
wis2box                 |   File "/usr/local/lib/python3.9/site-packages/requests/adapters.py", line 667, in send
wis2box                 |     resp = conn.urlopen(
wis2box                 |   File "/usr/local/lib/python3.9/site-packages/urllib3/connectionpool.py", line 873, in urlopen
wis2box                 |     return self.urlopen(
wis2box                 |   File "/usr/local/lib/python3.9/site-packages/urllib3/connectionpool.py", line 873, in urlopen
wis2box                 |     return self.urlopen(
wis2box                 |   File "/usr/local/lib/python3.9/site-packages/urllib3/connectionpool.py", line 873, in urlopen
wis2box                 |     return self.urlopen(
wis2box                 |   [Previous line repeated 1 more time]
wis2box                 |   File "/usr/local/lib/python3.9/site-packages/urllib3/connectionpool.py", line 843, in urlopen
wis2box                 |     retries = retries.increment(
wis2box                 |   File "/usr/local/lib/python3.9/site-packages/urllib3/util/retry.py", line 519, in increment
wis2box                 |     raise MaxRetryError(_pool, url, reason) from reason  # type: ignore[arg-type]
wis2box                 | urllib3.exceptions.MaxRetryError: HTTPConnectionPool(host='wis2box-api', port=80): Max retries exceeded with url: /oapi/admin/resources/ (Caused by NameResolutionError("<urllib3.connection.HTTPConnection object at 0xffff9c22eca0>: Failed to resolve 'wis2box-api' ([Errno -2] Name or service not known)"))
wis2box                 | 
wis2box                 | During handling of the above exception, another exception occurred:
wis2box                 | 
wis2box                 | Traceback (most recent call last):
wis2box                 |   File "/usr/local/bin/wis2box", line 33, in <module>
wis2box                 |     sys.exit(load_entry_point('wis2box==0.6.dev1', 'console_scripts', 'wis2box')())
wis2box                 |   File "/usr/local/lib/python3.9/site-packages/click/core.py", line 1157, in __call__
wis2box                 |     return self.main(*args, **kwargs)
wis2box                 |   File "/usr/local/lib/python3.9/site-packages/click/core.py", line 1078, in main
wis2box                 |     rv = self.invoke(ctx)
wis2box                 |   File "/usr/local/lib/python3.9/site-packages/click/core.py", line 1688, in invoke
wis2box                 |     return _process_result(sub_ctx.command.invoke(sub_ctx))
wis2box                 |   File "/usr/local/lib/python3.9/site-packages/click/core.py", line 1688, in invoke
wis2box                 |     return _process_result(sub_ctx.command.invoke(sub_ctx))
wis2box                 |   File "/usr/local/lib/python3.9/site-packages/click/core.py", line 1434, in invoke
wis2box                 |     return ctx.invoke(self.callback, **ctx.params)
wis2box                 |   File "/usr/local/lib/python3.9/site-packages/click/core.py", line 783, in invoke
wis2box                 |     return __callback(*args, **kwargs)
wis2box                 |   File "/usr/local/lib/python3.9/site-packages/click/decorators.py", line 33, in new_func
wis2box                 |     return f(get_current_context(), *args, **kwargs)
wis2box                 |   File "/usr/local/lib/python3.9/site-packages/wis2box-0.6.dev1-py3.9.egg/wis2box/api/__init__.py", line 149, in setup
wis2box                 |     if not api_config.has_collection(''):
wis2box                 |   File "/usr/local/lib/python3.9/site-packages/wis2box-0.6.dev1-py3.9.egg/wis2box/api/config/pygeoapi.py", line 95, in has_collection
wis2box                 |     r = self.http.get(f'{self.url}/{name}')
wis2box                 |   File "/usr/local/lib/python3.9/site-packages/requests/sessions.py", line 602, in get
wis2box                 |     return self.request("GET", url, **kwargs)
wis2box                 |   File "/usr/local/lib/python3.9/site-packages/requests/sessions.py", line 589, in request
wis2box                 |     resp = self.send(prep, **send_kwargs)
wis2box                 |   File "/usr/local/lib/python3.9/site-packages/requests/sessions.py", line 703, in send
wis2box                 |     r = adapter.send(request, **kwargs)
wis2box                 |   File "/usr/local/lib/python3.9/site-packages/requests/adapters.py", line 700, in send
wis2box                 |     raise ConnectionError(e, request=request)
wis2box                 | requests.exceptions.ConnectionError: HTTPConnectionPool(host='wis2box-api', port=80): Max retries exceeded with url: /oapi/admin/resources/ (Caused by NameResolutionError("<urllib3.connection.HTTPConnection object at 0xffff9c22eca0>: Failed to resolve 'wis2box-api' ([Errno -2] Name or service not known)"))
wis2box                 | Starting periodic command scheduler: cron.
wis2box                 | /etc/environment has been deprecated for locale information; use /etc/default/locale for LANG=C.UTF-8 instead ... (warning).
wis2box                 | cron is running.
wis2box                 | Setting up logger with level: ERROR
wis2box                 | Setting up logging (loglevel=ERROR, logfile=stdout)
wis2box                 | Setting up storage
wis2box                 | Creating baseline directory structure in /data/wis2box
wis2box                 | Setting up logger with level: ERROR
wis2box                 | WIS2BOX_HOST_DATADIR => /Users/cloftus/github/wis2box/hub2box-data
wis2box                 | WIS2BOX_DATADIR => /data/wis2box
wis2box                 | WIS2BOX_STORAGE_PUBLIC => hub2box-public
wis2box                 | WIS2BOX_BROKER_USERNAME => wis2box
wis2box                 | WIS2BOX_STORAGE_PASSWORD => minio_secret_key
wis2box                 | WIS2BOX_STORAGE_USERNAME => minio_access_key
wis2box                 | WIS2BOX_API_BACKEND_URL => http://frost-http:8080/FROST-Server/v1.1
wis2box                 | WIS2BOX_STORAGE_INCOMING => hub2box-incoming
wis2box                 | WIS2BOX_STORAGE_SOURCE => http://minio:9000
wis2box                 | WIS2BOX_LOGGING_LOGLEVEL => ERROR
wis2box                 | WIS2BOX_API_BACKEND_TYPE => SensorThings
wis2box                 | WIS2BOX_UI_CLUSTER => True
wis2box                 | WIS2BOX_BROKER_PASSWORD => wis2box
wis2box                 | WIS2BOX_BASEMAP_URL => https://{s}.tile.openstreetmap.org/{z}/{x}/{y}.png
wis2box                 | WIS2BOX_AUTH_URL => http://wis2box-auth
wis2box                 | WIS2BOX_API_URL => http://localhost:8999/oapi
wis2box                 | WIS2BOX_BROKER_HOST => mosquitto
wis2box                 | WIS2BOX_BROKER_PUBLIC => mqtt://wis2box:wis2box@mosquitto:1883
wis2box                 | WIS2BOX_DATA_RETENTION_DAYS => 30
wis2box                 | WIS2BOX_LOGGING_LOGFILE => stdout
wis2box                 | WIS2BOX_API_TYPE => pygeoapi
wis2box                 | WIS2BOX_BROKER_PORT => 1883
wis2box                 | WIS2BOX_BASEMAP_ATTRIBUTION => <a href="https://osm.org/copyright">OpenStreetMap</a> contributors
wis2box                 | WIS2BOX_STORAGE_DATA_RETENTION_DAYS => 7
wis2box                 | WIS2BOX_STORAGE_ARCHIVE => hub2box-archive
wis2box                 | WIS2BOX_DOCKER_API_URL => http://wis2box-api:80/oapi
wis2box                 | WIS2BOX_URL => http://localhost:8999
wis2box                 | WIS2BOX_STORAGE_TYPE => S3
wis2box                 | Setting up logger with level: ERROR

```

**Comment by C-Loftus** [comment]
Error again here

```
wis2box                 | Traceback (most recent call last):
wis2box                 |   File "/usr/local/lib/python3.9/site-packages/urllib3/connection.py", line 199, in _new_conn
wis2box                 |     sock = connection.create_connection(
wis2box                 |   File "/usr/local/lib/python3.9/site-packages/urllib3/util/connection.py", line 85, in create_connection
wis2box                 |     raise err
wis2box                 |   File "/usr/local/lib/python3.9/site-packages/urllib3/util/connection.py", line 73, in create_connection
wis2box                 |     sock.connect(sa)
wis2box                 | ConnectionRefusedError: [Errno 111] Connection refused
wis2box                 | 
wis2box                 | The above exception was the direct cause of the following exception:
wis2box                 | 
wis2box                 | Traceback (most recent call last):
wis2box                 |   File "/usr/local/lib/python3.9/site-packages/urllib3/connectionpool.py", line 789, in urlopen
wis2box                 |     response = self._make_request(
wis2box                 |   File "/usr/local/lib/python3.9/site-packages/urllib3/connectionpool.py", line 495, in _make_request
wis2box                 |     conn.request(
wis2box                 |   File "/usr/local/lib/python3.9/site-packages/urllib3/connection.py", line 441, in request
wis2box                 |     self.endheaders()
wis2box                 |   File "/usr/local/lib/python3.9/http/client.py", line 1280, in endheaders
wis2box                 |     self._send_output(message_body, encode_chunked=encode_chunked)
wis2box                 |   File "/usr/local/lib/python3.9/http/client.py", line 1040, in _send_output
wis2box                 |     self.send(msg)
wis2box                 |   File "/usr/local/lib/python3.9/http/client.py", line 980, in send
wis2box                 |     self.connect()
wis2box                 |   File "/usr/local/lib/python3.9/site-packages/urllib3/connection.py", line 279, in connect
wis2box                 |     self.sock = self._new_conn()
wis2box                 |   File "/usr/local/lib/python3.9/site-packages/urllib3/connection.py", line 214, in _new_conn
wis2box                 |     raise NewConnectionError(
wis2box                 | urllib3.exceptions.NewConnectionError: <urllib3.connection.HTTPConnection object at 0x7f1dec618a30>: Failed to establish a new connection: [Errno 111] Connection refused
wis2box                 | 
wis2box                 | The above exception was the direct cause of the following exception:
wis2box                 | 
wis2box                 | Traceback (most recent call last):
wis2box                 |   File "/usr/local/lib/python3.9/site-packages/requests/adapters.py", line 667, in send
wis2box                 |     resp = conn.urlopen(
wis2box                 |   File "/usr/local/lib/python3.9/site-packages/urllib3/connectionpool.py", line 873, in urlopen
wis2box                 |     return self.urlopen(
wis2box                 |   File "/usr/local/lib/python3.9/site-packages/urllib3/connectionpool.py", line 873, in urlopen
wis2box                 |     return self.urlopen(
wis2box                 |   File "/usr/local/lib/python3.9/site-packages/urllib3/connectionpool.py", line 873, in urlopen
wis2box                 |     return self.urlopen(
wis2box                 |   [Previous line repeated 1 more time]
wis2box                 |   File "/usr/local/lib/python3.9/site-packages/urllib3/connectionpool.py", line 843, in urlopen
wis2box                 |     retries = retries.increment(
wis2box                 |   File "/usr/local/lib/python3.9/site-packages/urllib3/util/retry.py", line 519, in increment
wis2box                 |     raise MaxRetryError(_pool, url, reason) from reason  # type: ignore[arg-type]
wis2box                 | urllib3.exceptions.MaxRetryError: HTTPConnectionPool(host='wis2box-api', port=80): Max retries exceeded with url: /oapi/admin/resources/ (Caused by NewConnectionError('<urllib3.connection.HTTPConnection object at 0x7f1dec618a30>: Failed to establish a new connection: [Errno 111] Connection refused'))
wis2box                 | 
wis2box                 | During handling of the above exception, another exception occurred:
wis2box                 | 
wis2box                 | Traceback (most recent call last):
wis2box                 |   File "/usr/local/bin/wis2box", line 33, in <module>
wis2box                 |     sys.exit(load_entry_point('wis2box==0.6.dev1', 'console_scripts', 'wis2box')())
wis2box                 |   File "/usr/local/lib/python3.9/site-packages/click/core.py", line 1157, in __call__
wis2box                 |     return self.main(*args, **kwargs)
wis2box                 |   File "/usr/local/lib/python3.9/site-packages/click/core.py", line 1078, in main
wis2box                 |     rv = self.invoke(ctx)
wis2box                 |   File "/usr/local/lib/python3.9/site-packages/click/core.py", line 1688, in invoke
wis2box                 |     return _process_result(sub_ctx.command.invoke(sub_ctx))
wis2box                 |   File "/usr/local/lib/python3.9/site-packages/click/core.py", line 1688, in invoke
wis2box                 |     return _process_result(sub_ctx.command.invoke(sub_ctx))
wis2box                 |   File "/usr/local/lib/python3.9/site-packages/click/core.py", line 1434, in invoke
wis2box                 |     return ctx.invoke(self.callback, **ctx.params)
wis2box                 |   File "/usr/local/lib/python3.9/site-packages/click/core.py", line 783, in invoke
wis2box                 |     return __callback(*args, **kwargs)
wis2box                 |   File "/usr/local/lib/python3.9/site-packages/click/decorators.py", line 33, in new_func
wis2box                 |     return f(get_current_context(), *args, **kwargs)
wis2box                 |   File "/usr/local/lib/python3.9/site-packages/wis2box-0.6.dev1-py3.9.egg/wis2box/api/__init__.py", line 149, in setup
wis2box                 |     if not api_config.has_collection(''):
wis2box                 |   File "/usr/local/lib/python3.9/site-packages/wis2box-0.6.dev1-py3.9.egg/wis2box/api/config/pygeoapi.py", line 95, in has_collection
wis2box                 |     r = self.http.get(f'{self.url}/{name}')
wis2box                 |   File "/usr/local/lib/python3.9/site-packages/requests/sessions.py", line 602, in get
wis2box                 |     return self.request("GET", url, **kwargs)
wis2box                 |   File "/usr/local/lib/python3.9/site-packages/requests/sessions.py", line 589, in request
wis2box                 |     resp = self.send(prep, **send_kwargs)
wis2box                 |   File "/usr/local/lib/python3.9/site-packages/requests/sessions.py", line 703, in send
wis2box                 |     r = adapter.send(request, **kwargs)
wis2box                 |   File "/usr/local/lib/python3.9/site-packages/requests/adapters.py", line 700, in send
wis2box                 |     raise ConnectionError(e, request=request)
wis2box                 | requests.exceptions.ConnectionError: HTTPConnectionPool(host='wis2box-api', port=80): Max retries exceeded with url: /oapi/admin/resources/ (Caused by NewConnectionError('<urllib3.connection.HTTPConnection object at 0x7f1dec618a30>: Failed to establish a new connection: [Errno 111] Connection refused'))
wis2box                 | /etc/environment has been deprecated for locale information; use /etc/default/locale for LANG=C.UTF-8 instead ... (warning).
wis2box                 | Starting periodic command scheduler: cron.
wis2box                 | /etc/environment has been deprecated for locale information; use /etc/default/locale for LANG=C.UTF-8 instead ... (warning).
wis2box                 | cron is running.
wis2box                 | Setting up logger with level: ERROR
wis2box                 | Setting up logging (loglevel=ERROR, logfile=stdout)
wis2box                 | Setting up storage
wis2box                 | Creating baseline directory structure in /data/wis2box
wis2box                 | Setting up logger with level: ERROR
wis2box                 | WIS2BOX_HOST_DATADIR => /home/cloftus/wis2box/hub2box-data
wis2box                 | WIS2BOX_DATADIR => /data/wis2box
wis2box                 | WIS2BOX_STORAGE_PUBLIC => hub2box-public
wis2box                 | WIS2BOX_BROKER_USERNAME => wis2box
wis2box                 | WIS2BOX_STORAGE_PASSWORD => minio_secret_key
wis2box                 | WIS2BOX_STORAGE_USERNAME => minio_access_key
wis2box                 | WIS2BOX_API_BACKEND_URL => http://frost-http:8080/FROST-Server/v1.1
wis2box                 | WIS2BOX_STORAGE_INCOMING => hub2box-incoming
wis2box                 | WIS2BOX_STORAGE_SOURCE => http://minio:9000
wis2box                 | WIS2BOX_LOGGING_LOGLEVEL => ERROR
wis2box                 | WIS2BOX_API_BACKEND_TYPE => SensorThings
wis2box                 | WIS2BOX_UI_CLUSTER => True
wis2box                 | WIS2BOX_BROKER_PASSWORD => wis2box
wis2box                 | WIS2BOX_BASEMAP_URL => https://{s}.tile.openstreetmap.org/{z}/{x}/{y}.png
wis2box                 | WIS2BOX_AUTH_URL => http://wis2box-auth
wis2box                 | WIS2BOX_API_URL => https://owdp-pilot.internetofwater.app/oapi
wis2box                 | WIS2BOX_BROKER_HOST => mosquitto
wis2box                 | WIS2BOX_BROKER_PUBLIC => mqtt://wis2box:wis2box@mosquitto:1883
wis2box                 | WIS2BOX_DATA_RETENTION_DAYS => 30
wis2box                 | WIS2BOX_LOGGING_LOGFILE => stdout
wis2box                 | WIS2BOX_API_TYPE => pygeoapi
wis2box                 | WIS2BOX_BROKER_PORT => 1883
wis2box                 | WIS2BOX_BASEMAP_ATTRIBUTION => <a href="https://osm.org/copyright">OpenStreetMap</a> contributors
wis2box                 | WIS2BOX_STORAGE_DATA_RETENTION_DAYS => 7
wis2box                 | WIS2BOX_STORAGE_ARCHIVE => hub2box-archive
wis2box                 | WIS2BOX_DOCKER_API_URL => http://wis2box-api:80/oapi
wis2box                 | WIS2BOX_URL => https://owdp-pilot.internetofwater.app
wis2box                 | WIS2BOX_STORAGE_TYPE => S3
wis2box                 | Setting up logger with level: ERROR
frost-mqtt              | 21:59:17.775 [            main] INFO           liquibase.lockservice - Waiting for changelog lock....
frost-http              | 21:59:20.504 [                main] INFO                liquibase.lockservice - Waiting for changelog lock....
frost-mqtt              | 21:59:27.793 [            main] INFO           liquibase.lockservice - Waiting for changelog lock....
frost-http              | 21:59:30.517 [                main] INFO                liquibase.lockservice - Waiting for changelog lock....
loki                    | level=info ts=2024-12-05T21:59:34.479733316Z caller=table_manager.go:171 msg="uploading tables"
loki                    | level=info ts=2024-12-05T21:59:34.479889766Z caller=table.go:335 msg="uploading table index_20062"
loki                    | level=info ts=2024-12-05T21:59:34.479913606Z caller=table.go:362 msg="finished uploading table index_20062"
loki                    | level=info ts=2024-12-05T21:59:34.479946046Z caller=table.go:420 msg="cleaning up unwanted dbs from table index_20062"
frost-mqtt              | 21:59:37.812 [            main] INFO           liquibase.lockservice - Waiting for changelog lock....
frost-http              | 21:59:40.533 [                main] INFO                liquibase.lockservice - Waiting for changelog lock....
wis2box                 | Traceback (most recent call last):
wis2box                 |   File "/usr/local/lib/python3.9/site-packages/urllib3/connection.py", line 199, in _new_conn
wis2box                 |     sock = connection.create_connection(
wis2box                 |   File "/usr/local/lib/python3.9/site-packages/urllib3/util/connection.py", line 85, in create_connection
wis2box                 |     raise err
wis2box                 |   File "/usr/local/lib/python3.9/site-packages/urllib3/util/connection.py", line 73, in create_connection
wis2box                 |     sock.connect(sa)
wis2box                 | ConnectionRefusedError: [Errno 111] Connection refused
wis2box                 | 
wis2box                 | The above exception was the direct cause of the following exception:
wis2box                 | 
wis2box                 | Traceback (most recent call last):
wis2box                 |   File "/usr/local/lib/python3.9/site-packages/urllib3/connectionpool.py", line 789, in urlopen
wis2box                 |     response = self._make_request(
wis2box                 |   File "/usr/local/lib/python3.9/site-packages/urllib3/connectionpool.py", line 495, in _make_request
wis2box                 |     conn.request(
wis2box                 |   File "/usr/local/lib/python3.9/site-packages/urllib3/connection.py", line 441, in request
wis2box                 |     self.endheaders()
wis2box                 |   File "/usr/local/lib/python3.9/http/client.py", line 1280, in endheaders
wis2box                 |     self._send_output(message_body, encode_chunked=encode_chunked)
wis2box                 |   File "/usr/local/lib/python3.9/http/client.py", line 1040, in _send_output
wis2box                 |     self.send(msg)
wis2box                 |   File "/usr/local/lib/python3.9/http/client.py", line 980, in send
wis2box                 |     self.connect()
wis2box                 |   File "/usr/local/lib/python3.9/site-packages/urllib3/connection.py", line 279, in connect
wis2box                 |     self.sock = self._new_conn()
wis2box                 |   File "/usr/local/lib/python3.9/site-packages/urllib3/connection.py", line 214, in _new_conn
wis2box                 |     raise NewConnectionError(
wis2box                 | urllib3.exceptions.NewConnectionError: <urllib3.connection.HTTPConnection object at 0x7f4ab536c250>: Failed to establish a new connection: [Errno 111] Connection refused
wis2box                 | 
wis2box                 | The above exception was the direct cause of the following exception:
wis2box                 | 
wis2box                 | Traceback (most recent call last):
wis2box                 |   File "/usr/local/lib/python3.9/site-packages/requests/adapters.py", line 667, in send
wis2box                 |     resp = conn.urlopen(
wis2box                 |   File "/usr/local/lib/python3.9/site-packages/urllib3/connectionpool.py", line 873, in urlopen
wis2box                 |     return self.urlopen(
wis2box                 |   File "/usr/local/lib/python3.9/site-packages/urllib3/connectionpool.py", line 873, in urlopen
wis2box                 |     return self.urlopen(
wis2box                 |   File "/usr/local/lib/python3.9/site-packages/urllib3/connectionpool.py", line 873, in urlopen
wis2box                 |     return self.urlopen(
wis2box                 |   [Previous line repeated 1 more time]
wis2box                 |   File "/usr/local/lib/python3.9/site-packages/urllib3/connectionpool.py", line 843, in urlopen
wis2box                 |     retries = retries.increment(
wis2box                 |   File "/usr/local/lib/python3.9/site-packages/urllib3/util/retry.py", line 519, in increment
wis2box                 |     raise MaxRetryError(_pool, url, reason) from reason  # type: ignore[arg-type]
wis2box                 | urllib3.exceptions.MaxRetryError: HTTPConnectionPool(host='wis2box-api', port=80): Max retries exceeded with url: /oapi/admin/resources/ (Caused by NewConnectionError('<urllib3.connection.HTTPConnection object at 0x7f4ab536c250>: Failed to establish a new connection: [Errno 111] Connection refused'))
wis2box                 | 
wis2box                 | During handling of the above exception, another exception occurred:
wis2box                 | 
wis2box                 | Traceback (most recent call last):
wis2box                 |   File "/usr/local/bin/wis2box", line 33, in <module>
wis2box                 |     sys.exit(load_entry_point('wis2box==0.6.dev1', 'console_scripts', 'wis2box')())
wis2box                 |   File "/usr/local/lib/python3.9/site-packages/click/core.py", line 1157, in __call__
wis2box                 |     return self.main(*args, **kwargs)
wis2box                 |   File "/usr/local/lib/python3.9/site-packages/click/core.py", line 1078, in main
wis2box                 |     rv = self.invoke(ctx)
wis2box                 |   File "/usr/local/lib/python3.9/site-packages/click/core.py", line 1688, in invoke
wis2box                 |     return _process_result(sub_ctx.command.invoke(sub_ctx))
wis2box                 |   File "/usr/local/lib/python3.9/site-packages/click/core.py", line 1688, in invoke
wis2box                 |     return _process_result(sub_ctx.command.invoke(sub_ctx))
wis2box                 |   File "/usr/local/lib/python3.9/site-packages/click/core.py", line 1434, in invoke
wis2box                 |     return ctx.invoke(self.callback, **ctx.params)
wis2box                 |   File "/usr/local/lib/python3.9/site-packages/click/core.py", line 783, in invoke
wis2box                 |     return __callback(*args, **kwargs)
wis2box                 |   File "/usr/local/lib/python3.9/site-packages/click/decorators.py", line 33, in new_func
wis2box                 |     return f(get_current_context(), *args, **kwargs)
wis2box                 |   File "/usr/local/lib/python3.9/site-packages/wis2box-0.6.dev1-py3.9.egg/wis2box/api/__init__.py", line 149, in setup
wis2box                 |     if not api_config.has_collection(''):
wis2box                 |   File "/usr/local/lib/python3.9/site-packages/wis2box-0.6.dev1-py3.9.egg/wis2box/api/config/pygeoapi.py", line 95, in has_collection
wis2box                 |     r = self.http.get(f'{self.url}/{name}')
wis2box                 |   File "/usr/local/lib/python3.9/site-packages/requests/sessions.py", line 602, in get
wis2box                 |     return self.request("GET", url, **kwargs)
wis2box                 |   File "/usr/local/lib/python3.9/site-packages/requests/sessions.py", line 589, in request
wis2box                 |     resp = self.send(prep, **send_kwargs)
wis2box                 |   File "/usr/local/lib/python3.9/site-packages/requests/sessions.py", line 703, in send
wis2box                 |     r = adapter.send(request, **kwargs)
wis2box                 |   File "/usr/local/lib/python3.9/site-packages/requests/adapters.py", line 700, in send
wis2box                 |     raise ConnectionError(e, request=request)
wis2box                 | requests.exceptions.ConnectionError: HTTPConnectionPool(host='wis2box-api', port=80): Max retries exceeded with url: /oapi/admin/resources/ (Caused by NewConnectionError('<urllib3.connection.HTTPConnection object at 0x7f4ab536c250>: Failed to establish a new connection: [Errno 111] Connection refused'))
wis2box                 | /etc/environment has been deprecated for locale information; use /etc/default/locale for LANG=C.UTF-8 instead ... (warning).
wis2box                 | Starting periodic command scheduler: cron.
wis2box                 | /etc/environment has been deprecated for locale information; use /etc/default/locale for LANG=C.UTF-8 instead ... (warning).
wis2box                 | cron is running.
wis2box                 | Setting up logger with level: ERROR
wis2box                 | Setting up logging (loglevel=ERROR, logfile=stdout)
wis2box                 | Setting up storage
wis2box                 | Creating baseline directory structure in /data/wis2box
web-proxy               | 172.18.0.1 - - [05/Dec/2024:21:59:46 +0000] "GET /oapi/collections/things/items HTTP/1.1" 502 157 "https://owdp-pilot.internetofwater.app/" "Mozilla/5.0 (Macintosh; Intel Mac OS X 10.15; rv:133.0) Gecko/20100101 Firefox/133.0"
web-proxy               | 2024/12/05 21:59:46 [error] 22#22: *1 connect() failed (111: Connection refused) while connecting to upstream, client: 172.18.0.1, server: , request: "GET /oapi/collections/things/items HTTP/1.1", upstream: "http://172.18.0.11:80/oapi/collections/things/items", host: "owdp-pilot.internetofwater.app", referrer: "https://owdp-pilot.internetofwater.app/"
wis2box                 | Setting up logger with level: ERROR
wis2box                 | WIS2BOX_HOST_DATADIR => /home/cloftus/wis2box/hub2box-data
wis2box                 | WIS2BOX_DATADIR => /data/wis2box
wis2box                 | WIS2BOX_STORAGE_PUBLIC => hub2box-public
wis2box                 | WIS2BOX_BROKER_USERNAME => wis2box
wis2box                 | WIS2BOX_STORAGE_PASSWORD => minio_secret_key
wis2box                 | WIS2BOX_STORAGE_USERNAME => minio_access_key
wis2box                 | WIS2BOX_API_BACKEND_URL => http://frost-http:8080/FROST-Server/v1.1
wis2box                 | WIS2BOX_STORAGE_INCOMING => hub2box-incoming
wis2box                 | WIS2BOX_STORAGE_SOURCE => http://minio:9000
wis2box                 | WIS2BOX_LOGGING_LOGLEVEL => ERROR
wis2box                 | WIS2BOX_API_BACKEND_TYPE => SensorThings
wis2box                 | WIS2BOX_UI_CLUSTER => True
wis2box                 | WIS2BOX_BROKER_PASSWORD => wis2box
wis2box                 | WIS2BOX_BASEMAP_URL => https://{s}.tile.openstreetmap.org/{z}/{x}/{y}.png
wis2box                 | WIS2BOX_AUTH_URL => http://wis2box-auth
wis2box                 | WIS2BOX_API_URL => https://owdp-pilot.internetofwater.app/oapi
wis2box                 | WIS2BOX_BROKER_HOST => mosquitto
wis2box                 | WIS2BOX_BROKER_PUBLIC => mqtt://wis2box:wis2box@mosquitto:1883
wis2box                 | WIS2BOX_DATA_RETENTION_DAYS => 30
wis2box                 | WIS2BOX_LOGGING_LOGFILE => stdout
wis2box                 | WIS2BOX_API_TYPE => pygeoapi
wis2box                 | WIS2BOX_BROKER_PORT => 1883
wis2box                 | WIS2BOX_BASEMAP_ATTRIBUTION => <a href="https://osm.org/copyright">OpenStreetMap</a> contributors
wis2box                 | WIS2BOX_STORAGE_DATA_RETENTION_DAYS => 7
wis2box                 | WIS2BOX_STORAGE_ARCHIVE => hub2box-archive
wis2box                 | WIS2BOX_DOCKER_API_URL => http://wis2box-api:80/oapi
wis2box                 | WIS2BOX_URL => https://owdp-pilot.internetofwater.app
wis2box                 | WIS2BOX_STORAGE_TYPE => S3
wis2box                 | Setting up logger with level: ERROR
frost-mqtt              | 21:59:47.827 [            main] INFO           liquibase.lockservice 
```

**Comment by C-Loftus** [comment]
Fixed with #10
